templates/FFV_form_factors.cpp.in

```diff
@@ -239,12 +365,214 @@ double OneLoopFunctionF(double r)
               0.1875000000000000000  * y * y * y * y * y * y * y * y * y * y * y * y * y +
               0.1764705882352941176  * y * y * y * y * y * y * y * y * y * y * y * y * y * y -
               0.16666666666666666667 * y * y * y * y * y * y * y * y * y * y * y * y * y * y * y);
+   } else if (1./r < ratiolimit) {
+      // Expand about 0 to 1st order
+      return 1.5*r;
    } else {
       return 3. / 2. * (-3.0 + 4.0 * r - r * r - 2.0 * std::log(r)) /
-             pow(1.0 - r, 3);
+             Power3(1.0 - r);
+   }
+}
+
+double OneLoopFunctionH(double r)
+{
+   if (is_zero(r)) {
+      return 5.0;
+   } else if (is_zero(1.0 - r)){
+      return 0.0;
+   } else if (1./r < Power3(ratiolimit)) {
+      // Loop function is about 10^-13, treat as reaching the 0 limit
+      return 0.;
+   } else if (1./r < ratiolimit) {
+      // Expand about 0 to 1st order
+      return r*(-5.-6.*std::log(r));
+   } else {
+      return (5.0 - 27.0 * r + 27.0 * r * r - 5.0 * r * r * r + 6.0 * (r - 3.0) * r * r * std::log(r)) /
+              Power4(1.0 - r);
+   }
+}
+
+double OneLoopFunctionI(double r)
```

Comment:
Please add a reference for this function.

Reply:
There are none that I know of, especially in this gauge (Feynman).  For the other vector one loop function I was able to refer to the paper Peter, Wojtek, and I just put on arxiv, but this one loop function is not in there as it is not needed for muon g-2.

Reply:
Ok, I see.

Will there be a future reference? Maybe in a follow-up paper or so? If so, we should come back and add a comment here. What do you think?

Reply:
I don't think so for these functions in particular, they are not that relevant for g-2 unfortunately.

Reply:
Hmm, I see.

I just have the feeling that it would be good to have some kind of more descriptive documentation of these functions, beyond the implementation themselves. For example, some Feynman diagram correspondence or some integral formula etc. Just something where one could start if one wishes to re-derive or think about these functions later again. (Are you sure that you don't want to write a small publication with these calculations? ;) )  What about adding a more detailed Doxygen-style comment that explains the Feynman diagram and/or the integral corresponding to these functions?

Reply:
@Expander, no worries, I will get on to writing an explanation for them.

Reply:
Many thanks!  :)

Reply:
Some brief explanations are there now for the simple one-loop functions, for every one that does not have a literature reference.